eng/common/scripts/get-codeowners.ps1

```diff
@@ -0,0 +1,43 @@
+param (
+  $TargetDirectory, # should be in relative form from root of repo. EG: sdk/servicebus
+  $RootDirectory # ideally $(Build.SourcesDirectory)
+)
+
+$codeOwnersLocation = @(Get-ChildItem -R -Force -Path $RootDirectory -Filter "CODEOWNERS")
+
+if ($codeOwnersLocation.Length -eq 0) {
+  Write-Host "Unable to find CODEOWNERS file in target directory $RootDirectory"
+  exit 1
+}
+
+if ($codeOwnersLocation.Length -gt 1) {
+  Write-Host "Multiple CODEOWNERS files detected in  $RootDirectory."
+  Write-Host "$codeOwnersLocation"
+  exit 1
+}
+
+
+$codeOwnersContent = Get-Content ($codeOwnersLocation | Select-Object -Last 1)
+
+$ownedFolders = @{}
+
+foreach ($contentLine in $codeOwnersContent) {
+  if (-not $contentLine.StartsWith("#") -and $contentLine){
+    $splitLine = $contentLine -split "\s" | ? { return $_ }
+    
+    # in the codeowners file, gh aliases start with @. we don't want that when passing them to the API
+    $ownedFolders[$splitLine[0].ToLower()] = ($splitLine[1..$($splitLine.Length)] | % { return $_.substring(1) }) -join ","
+  }
+}
+
+$results = $ownedFolders[$TargetDirectory.ToLower()]
```

Comment:
This assumes that the target directories exactly match (ignoring case) what is in the CODEOWNERS file. That means the caller is going to have to try have some knowledge of how to correctly format this to get the results they want. We may need to think a little more about how to normalize these paths. I wouldn't go as far as evaluating wildcards but perhaps some logic for finding the closest matching path.

Reply:
Follow-up issue being filed. This could be super finicky, and the way it's currently written will definitely work with the service specific codeowners stuff.

Reply:
https://github.com/Azure/azure-sdk-tools/issues/784